diff --git a/src/Music/Time/Note.hs b/src/Music/Time/Note.hs
index f08a93e..745f051 100644
--- a/src/Music/Time/Note.hs
+++ b/src/Music/Time/Note.hs
@@ -131,6 +131,10 @@ instance Splittable a => Splittable (Note a) where
 instance Reversible (Note a) where
   rev = revDefault
 
+instance Inside Note where
+  ins = noteValue
+
+
 -- |
 -- View a note as a pair of the original value and the transformation (and vice versa).
 --
diff --git a/src/Music/Time/Transform.hs b/src/Music/Time/Transform.hs
index 2adc1e7..6cf61f8 100644
--- a/src/Music/Time/Transform.hs
+++ b/src/Music/Time/Transform.hs
@@ -55,10 +55,16 @@ module Music.Time.Transform (
       stretching,
       compressing,
 
+      -- * Inside class
+      Inside(..),
+
       -- * Utility
       delayTime,
   ) where
 
 import           Music.Time.Internal.Transform
 import           Music.Time.Types
+import Control.Lens (Lens(..))
 
+class Inside f where
+  ins :: (Transformable a, Transformable b) => Lens (f a) (f b) a b
